File: 2022-2023/sourcecode/lab/lab9/RoomRelationsJava/app/src/main/java/pl/edu/uwr/pum/roomrelationsjava/model/entities/relations/FacultyAndDean.java
package pl.edu.uwr.pum.roomrelationsjava.model.entities.relations;

import androidx.room.Embedded;
import androidx.room.Relation;

import pl.edu.uwr.pum.roomrelationsjava.model.entities.Dean;
import pl.edu.uwr.pum.roomrelationsjava.model.entities.Faculty;

public class FacultyAndDean {
    @Embedded
    public Faculty faculty;

    @Relation(
            parentColumn = "facultyName",
            entityColumn = "facultyName"
    )

    public Dean dean;
}

File: 2022-2023/sourcecode/lab/lab9/RoomRelationsKotlin/app/src/main/java/pl/edu/uwr/pum/roomrelationskotlin/model/entity/relations/FacultyAndDean.kt
package pl.edu.uwr.pum.roomrelationskotlin.model.entity.relations

import androidx.room.Embedded
import androidx.room.Relation
import pl.edu.uwr.pum.roomrelationskotlin.model.entity.Dean
import pl.edu.uwr.pum.roomrelationskotlin.model.entity.Faculty

data class FacultyAndDean (
    @Embedded val faculty: Faculty,

    @Relation(
        parentColumn = "facultyName",
        entityColumn = "facultyName"
    )

    val dean: Dean
    )